### PR TITLE
Add page that draws a raffle from paid, registered members.

### PR DESCRIPTION
**Pull request by katiewalker**
Adds a page that selects an individual at random from the group of people who've registered, and paid.

**Side Effects**
- move Firebase initialisation out of `join.html`, into `csclub.common.js`.
- authentication is a thing!

**Concurrent Changes**
- security rules for the Firebase database
  - default is no access
  - anyone can write to `members/`
  - only people logged in with a `@csclub.org.au` account can write

**Notes**
If we use authentication elsewhere on the site, the logic can be moved out of `csclub.raffle.js` without affecting anything.

### CODE REVIEW

**Comment by katiewalker** [comment]
Preview here https://rawgit.com/compsci-adl/compsci-adl.github.io/raffle/raffle.html

File: js/csclub.raffle.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    while (!haveWinner) {
+      var i = 0;
+      var rand = Math.floor(Math.random() * snapshot.numChildren());
+      snapshot.forEach(function(snapshot) {
+        if (i == rand) {
+          if (snapshot.val().square_receipt.length > 0) {
+            haveWinner = true;
+            writeWinner(snapshot.val().first_name + " " + snapshot.val().last_name);
+          } 
+          break;
```
Comment (JakeLane) [inline js/csclub.raffle.js]: There is no `break` in the forEach method. Either omit or use a different construct.

File: js/csclub.raffle.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        }
+        i++;
+      });
+    }
+  }).catch(function(error) {
+    writeWinner("Sorry, you don't have permission to do this."); 
+  });
+
+  console.log(name);
```
Comment (JakeLane) [inline js/csclub.raffle.js]: Unnecessary `console.log`

**Review by JakeLane — CHANGES_REQUESTED** [review]

File: js/csclub.raffle.js (hunk 1 above)
Comment (katiewalker) [reply js/csclub.raffle.js]: boo JS.

File: js/csclub.raffle.js (hunk 2 above)
Comment (katiewalker) [reply js/csclub.raffle.js]: Done.

**Review by JakeLane — APPROVED** [review]

**Pull request merged**